### PR TITLE
upgrade: Improve various error messages

### PR DESCRIPTION
**Pull request by skazi0**
Multiple error handling/message improvements. Please check commit messages for details.

Includes https://github.com/crowbar/crowbar-core/pull/1585

### CODE REVIEW

**Comment by jsuchome** [comment]
@skazi0 would you mind making this a bigger cleanup PR, adding also 

- https://github.com/crowbar/crowbar-core/pull/1585/commits/2a59894eb09369cb278cb519a3ed4388ac15cfb4

- http://pastebin.nue.suse.com/21339/src

**Comment by skazi0** [comment]
@jsuchome  sure I can include the change from pastebin but the commit you linked is already in the repo, isn't it?

**Comment by jsuchome** [comment]
I don't think so. I've just created https://github.com/crowbar/crowbar-core/pull/1585  some time ago and did not want to merge it yet as it was very little change

File: crowbar_framework/app/models/node.rb [hunk 1]
```diff
@@ -903,6 +903,8 @@ def wait_for_script_to_finish(script, seconds, args = [])
     cmd = script
     cmd += " " + args.join(" ") unless args.empty?
 
+    raise "Script #{script} is not present at the node #{@node.name}." unless file_exist?(script)
```
Comment (rhafer) [inline crowbar_framework/app/models/node.rb]: I think it's `s/at/on/` but you might want to check back with a native speaker.

Comment (skazi0) [reply crowbar_framework/app/models/node.rb]: I would probably say "on" myself too, but I stole this message from @jsuchome :)

Comment (jgravel) [reply crowbar_framework/app/models/node.rb]: I've asked Gary Smith who's a native speaker :-) He recommended using 'on' instead of 'at' in this case.

Comment (jsuchome) [reply crowbar_framework/app/models/node.rb]: Oh, thanks for assessing my English ;-)

(Feel free to correct it of course)

**Pull request merged**